src/tests/quantuminspire/qiskit/test_measurements.py

```diff
@@ -0,0 +1,220 @@
+
+    def test_validate_nr_classical_qubits_less_than_needed_for_storing_measured_qubits(self):
+        q = QuantumRegister(2, "q")
+        c = ClassicalRegister(1, "c")
+        qc = QuantumCircuit(q, c, name="conditional")
+        qc.cx(q[0], q[1])
+
+        experiment = self._circuit_to_experiment(qc)
+        self.assertRaisesRegex(QiskitBackendError, 'Number of classical bits \(1\) is not sufficient for storing the '
+                                                   'outcomes of the experiment',
+                               self.measurements.collect_measurements, experiment)
+
+    def test_invalid_amount_of_classical_bits(self):
+        qc = QuantumCircuit(2, 2)
+        qc.measure(1, 0)
+
+        qobj = self._circuit_to_qobj(qc)
+        qobj.experiments[0].header.memory_slots = 0
+
+        experiment = qobj.experiments[0]
+        self.assertRaisesRegex(QiskitBackendError, 'Invalid amount of classical bits \(0\)!',
+                               self.measurements.collect_measurements, experiment)
+
+    def test_max_measurement_index(self):
+        qr = QuantumRegister(5)
+        cr = ClassicalRegister(3)
+        cr_ghz = ClassicalRegister(2)
+        circuit = QuantumCircuit(qr, cr, cr_ghz)
+
+        circuit.measure(2, 3)
+        circuit.measure(3, 4)
+        circuit.measure([0, 1, 4], cr)
+        circuit.measure([2, 3], cr_ghz)
+
+        experiment = self._circuit_to_experiment(circuit)
+        self.measurements.collect_measurements(experiment)
+
+        self.assertEqual(self.measurements.max_measurement_index, 4)
+
+    def test_max_measurement_index_less_than_nr_of_clbits(self):
+        qr = QuantumRegister(5)
+        cr = ClassicalRegister(3)
+        cr_ghz = ClassicalRegister(2)
+        circuit = QuantumCircuit(qr, cr, cr_ghz)
+
+        circuit.measure(2, 3)
+        circuit.measure([0, 1, 4], cr)
+        circuit.measure([2], cr_ghz[0])
+
+        experiment = self._circuit_to_experiment(circuit)
+        self.measurements.collect_measurements(experiment)
+
+        self.assertEqual(self.measurements.max_measurement_index, 3)
+
+    def test_get_qreg_for_conditional_creg(self):
+        qr = QuantumRegister(5)
+        cr = ClassicalRegister(3)
+        cr_ghz = ClassicalRegister(2)
+        circuit = QuantumCircuit(qr, cr, cr_ghz)
+
+        circuit.measure(2, 3)
+        circuit.measure([1, 4], [1, 2])
+        circuit.measure([2], cr_ghz[0])
+
+        experiment = self._circuit_to_experiment(circuit)
+        self.measurements.collect_measurements(experiment)
+
+        self.assertEqual(self.measurements.get_qreg_for_conditional_creg(3), 2)
+        self.assertEqual(self.measurements.get_qreg_for_conditional_creg(1), 1)
+        self.assertEqual(self.measurements.get_qreg_for_conditional_creg(2), 4)
+        self.assertEqual(self.measurements.get_qreg_for_conditional_creg(0), 0)
+        self.assertRaisesRegex(QiskitBackendError, "Classical bit 4 used in a conditional gate is not measured and the "
```

Comment:
nit: single quotes are used in another `assertRaisesRegex` call above and also one below. Would be good to be consistent within the file.

Reply:
ok